src/node/SetupServerApi.ts

```diff
@@ -115,6 +129,27 @@ export class SetupServerApi
     )
   }
 
+  private async applyRequestHandling(
+    request: InteractiveRequest,
+    requestId: string,
+    handlers: Array<RequestHandler>,
+    options: RequiredDeep<SharedOptions>,
+    emitter: Emitter<LifeCycleEventsMap>,
+  ) {
+    const response = await handleRequest(
+      request,
+      requestId,
+      handlers,
+      options,
+      emitter,
+    )
+
+    if (response) {
+      request.respondWith(response)
+    }
+    return
+  }
+
```

Comment:
I think this could be tidier, but I'm not sure how to best place/refactor it. It's used twice above, and requires `this` to be up-to-date for the handlers etc.
This requires a type for request, but could be changed to `any` if you preferred to avoid defining the interface (with comment above).